modules/rostests/apitests/ntdll/DllLoadNotification.c

```diff
@@ -0,0 +1,225 @@
+/*
+ * PROJECT:     ReactOS API Tests
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Test for DLL Load Notification API
+ * COPYRIGHT:   Copyright 2024 Ratin Gao <ratin@knsoft.org>
+ */
+
+#define UNICODE
+
+#include "precomp.h"
+
+#include <winuser.h>
+
+static WCHAR g_szDllPath[MAX_PATH];
+static UNICODE_STRING g_usDllPath;
+static UNICODE_STRING g_usDllName;
+static volatile LONG g_lDllLoadCount = 0;
+
+typedef
+NTSTATUS
+NTAPI
+FN_LdrRegisterDllNotification(
+    _In_ ULONG Flags,
+    _In_ PLDR_DLL_NOTIFICATION_FUNCTION NotificationFunction,
+    _In_opt_ PVOID Context,
+    _Out_ PVOID* Cookie);
+
+typedef
+NTSTATUS
+NTAPI
+FN_LdrUnregisterDllNotification(
+    _In_ PVOID Cookie);
+
+static BOOL ExtractResource(
+    _In_z_ PCWSTR SavePath,
+    _In_ PCWSTR ResourceType,
+    _In_ PCWSTR ResourceName)
+{
+    BOOL bSuccess;
+    DWORD dwWritten, dwSize;
+    HGLOBAL hGlobal;
+    LPVOID pData;
+    HANDLE hFile;
+    HRSRC hRsrc;
+
+    /* Load resource */
+    if ((hRsrc = FindResourceW(NULL, ResourceName, ResourceType)) == NULL ||
+        (dwSize = SizeofResource(NULL, hRsrc)) == 0 ||
+        (hGlobal = LoadResource(NULL, hRsrc)) == NULL ||
+        (pData = LockResource(hGlobal)) == NULL)
+    {
+        return FALSE;
+    }
+
+    /* Save to file */
+    hFile = CreateFileW(SavePath, GENERIC_WRITE, FILE_SHARE_READ, NULL, CREATE_ALWAYS, FILE_ATTRIBUTE_NORMAL, NULL);
+    if (hFile == INVALID_HANDLE_VALUE)
+    {
+        return FALSE;
+    }
+    bSuccess = WriteFile(hFile, pData, dwSize, &dwWritten, NULL);
+    CloseHandle(hFile);
+    if (!bSuccess)
+    {
+        return FALSE;
+    } else if (dwWritten != dwSize)
+    {
+        trace("Extract resource failed, written size (%lu) is not actual size (%lu).\n", dwWritten, dwSize);
```

Comment:
```suggestion
        trace("Extract resource failed, written size (%lu) is not actual size (%lu)\n", dwWritten, dwSize);
```

Reply:
Feel free to ignore nitpicking like this, not everyone cares about 1 saved byte.